### PR TITLE
allow IDT endpoint to upload documents associated with legacy appeals

### LINKED ISSUE #14971: Allow document uploads to be connected with legacy appeals as well as AMA appeals
Follows #9567

## Description
As an attorney using IDT, I want to upload documents associated with any type of appeal that I may be working on.

Currently, the `upload_document` endpoint accepts only AMA appeal uuids. Because IDT's letter creator feature has expanded in scope recently to handle a need for a large volume of legacy letters, we should expand the endpoint's capabilities to accept legacy appeal VACOLS ids as well.

## Acceptance criteria
- [ ] IDT users can upload documents associated with legacy appeals by passing the appeal's VACOLS id
- [ ] controller and model tests for the new functionality

## Background/context/resources
[Slack thread](https://dsva.slack.com/archives/CD5DAQNCU/p1597325742020200) where the problem was discovered.

### PR DESCRIPTION
**Pull request by tomas-nava**
Resolves #14971
Follows #14972

### Description
Modifies the `VbmsUploadedDocument` to have a polymorphic appeal relationship, so that uploads can be associated with either AMA appeals or legacy appeals. Updates related files and tests.

### Acceptance criteria
- [x] **BEFORE MERGING THIS PR**, manually update data in production to make sure there is no `VbmsUploadedDocument` object with an empty `appeal_type` column. Because all current uploads are associated with AMA appeals by default, this can be done by running this code in the production console:
  ```ruby
  VbmsUploadedDocument.where(appeal_type: nil).in_batches.update_all(appeal_type: Appeal.name)
  ```

### Database Changes
* ~~Timestamps (created_at, updated_at) for new tables~~ n/a
* [x] Column comments updated
* [x] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [x] Query profiling performed (eyeball Rails log, check bullet and fasterer output)
* [x] Appropriate indexes added (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [x] DB schema docs updated with `make docs` (after running `make migrate`)
* [x] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

File: app/workflows/prepare_document_upload_to_vbms.rb [hunk 1]
```diff
@@ -28,8 +28,8 @@ def call
 
   attr_reader :document_type, :file, :params, :success
 
-  def appeal_id
-    Appeal.find_by(uuid: params[:appeal_id])&.id
+  def appeal
+    @appeal ||= Appeal.find_appeal_by_id_or_find_or_create_legacy_appeal_by_vacols_id(params[:appeal_id])
```
Comment (ferristseng) [inline app/workflows/prepare_document_upload_to_vbms.rb:32]: So the `appeal_id` here is kind of like what we think of as the `external_id`?

File: app/models/vbms_uploaded_document.rb [hunk 2]
```diff
@@ -1,22 +1,14 @@
 # frozen_string_literal: true
 
 class VbmsUploadedDocument < CaseflowRecord
-  include HasSimpleAppealUpdatedSince
+  include HasAppealUpdatedSince
 
-  belongs_to :appeal, optional: false
+  belongs_to :appeal, polymorphic: true
```
[the remark this answers is not in the document]
Comment (ferristseng) [reply app/models/vbms_uploaded_document.rb:6]: How does this know that for `Appeal` the relationship is `appeal_id => uuid` and for `LegacyAppeal` the relationship is `appeal_id => vacols_id` (like what happens in `PrepareDocumentUploadToVbms`)?

[the remark this answers is not in the document]
Comment (tomas-nava) [reply app/models/vbms_uploaded_document.rb:6]: According to [the rails guide](https://guides.rubyonrails.org/association_basics.html#optional), `optional` is `false` by default, so I think it's safe to remove.

File: app/workflows/prepare_document_upload_to_vbms.rb (hunk 1 above)
Comment (tomas-nava) [reply app/workflows/prepare_document_upload_to_vbms.rb:32]: yes, [`find_appeal_by_id_or_find_or_create_legacy_appeal_by_vacols_id`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/appeal.rb#L76) takes an `Appeal`'s `uuid` or a `LegacyAppeal`'s `vacols_id`. We should probably rename it to `find_appeal_by_uuid_or...`

File: app/models/vbms_uploaded_document.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (tomas-nava) [reply app/models/vbms_uploaded_document.rb:6]: It uses the `uuid` or `vacols_id` [to find/create the appeal](https://github.com/department-of-veterans-affairs/caseflow/pull/14995/files#diff-f6565390888bd450349acb72bd7a0d1aR32), but when it creates the `VbmsUploadedDocument`, it [uses the appeal's database id](https://github.com/department-of-veterans-affairs/caseflow/pull/14995/files#diff-f6565390888bd450349acb72bd7a0d1aR41).

[the remark this answers is not in the document]
Comment (ferristseng) [reply app/models/vbms_uploaded_document.rb:6]: Gotcha! That makes sense 🧠

**Comment by tomas-nava** [comment]
I updated the `VbmsUploadedDocument` records in production:
```ruby
> VbmsUploadedDocument.where(appeal_type: nil).count
=> 21444
> VbmsUploadedDocument.where(appeal_type: Appeal.name).count
=> 45
> VbmsUploadedDocument.where(appeal_type: nil).in_batches.update_all(appeal_type: Appeal.name)
> VbmsUploadedDocument.where(appeal_type: nil).count
=> 0
> VbmsUploadedDocument.where(appeal_type: Appeal.name).count
=> 21489
```

**Comment by tomas-nava** [comment]
I updated the `VbmsUploadedDocument` records in UAT as well:

```ruby
> VbmsUploadedDocument.where(appeal_type: nil).count
=> 9
> VbmsUploadedDocument.where(appeal_type: Appeal.name).count
=> 0
> VbmsUploadedDocument.where(appeal_type: nil).in_batches.update_all(appeal_type: Appeal.name)
> VbmsUploadedDocument.where(appeal_type: nil).count
=> 0
> VbmsUploadedDocument.where(appeal_type: Appeal.name).count
=> 9
```

**Pull request merged**